infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -227,72 +298,39 @@ private void multiRead(CorfuPayloadMsg<MultipleReadRequest> msg, ChannelHandlerC
-        } catch (DataOutrankedException e) {
-            r.sendResponse(ctx, msg, CorfuMsgType.ERROR_DATA_OUTRANKED.msg());
-        } catch (ValueAdoptedException e) {
-
-            r.sendResponse(ctx, msg, CorfuMsgType.ERROR_VALUE_ADOPTED.payloadMsg(e
-                    .getReadResponse()));
-        }
-    }
-
-    @ServerHandler(type = CorfuMsgType.PREFIX_TRIM)
-    private void prefixTrim(CorfuPayloadMsg<TrimRequest> msg, ChannelHandlerContext ctx,
-                            IServerRouter r) {
-        try {
-            TrimRequest req = msg.getPayload();
-            batchWriter.prefixTrim(req.getAddress());
-            r.sendResponse(ctx, msg, CorfuMsgType.ACK.msg());
-        } catch (TrimmedException ex) {
-            r.sendResponse(ctx, msg, CorfuMsgType.ERROR_TRIMMED.msg());
-        }
-    }
-
     @ServerHandler(type = CorfuMsgType.COMPACT_REQUEST)
-    private void compact(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
-        try {
-            streamLog.compact();
-            r.sendResponse(ctx, msg, CorfuMsgType.ACK.msg());
-        } catch (RuntimeException ex) {
-            log.error("Internal Error");
-            //TODO(Maithem) Need an internal error return type
-            r.sendResponse(ctx, msg, CorfuMsgType.ACK.msg());
-        }
+    private void handleCompactRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+        log.debug("handleCompactRequest: received a compact request {}", msg);
+        streamLog.compact();
+        r.sendResponse(ctx, msg, CorfuMsgType.ACK.msg());
     }
 
     @ServerHandler(type = CorfuMsgType.FLUSH_CACHE)
-    private void flushCache(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
-        try {
-            dataCache.invalidateAll();
-        } catch (RuntimeException e) {
-            log.error("Encountered error while flushing cache {}", e);
-        }
+    private void handleFlushCacheRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+        log.debug("handleFlushCacheRequest: received a cache flush request {}", msg);
+        dataCache.invalidateAll();
         r.sendResponse(ctx, msg, CorfuMsgType.ACK.msg());
     }
 
     /**
      * Seal the server with the epoch.
-     *
+     * <p>
      * - A seal operation is inserted in the queue and then we wait to flush all operations
-     *   in the queue before this operation.
+     * in the queue before this operation.
      * - All operations after this operation but stamped with an older epoch will be failed.
      */
     @Override
     public void sealServerWithEpoch(long epoch) {
-        batchWriter.waitForSealComplete(epoch);
+        CorfuPayloadMsg<Long> msg = new CorfuPayloadMsg<>();
+        msg.setEpoch(epoch);
+        //TODO(Maithem): need to unwrap CompletionException for WrongEpochExceptions etc...
+        try {
+            batchWriter.addTask(SEAL, msg).join();
```

Comment:
We need to wait on the future.

Reply:
`CompletableFuture::join`

Is a blocking call.